pkg/danmep/calico.go

```diff
@@ -0,0 +1,36 @@
+package danmep
+
+import (
+    "context"
+    "fmt"
+    "log"
+    "net"
+    "os/exec"
+
+    danmtypes "github.com/nokia/danm/crd/apis/danm/v1"
+    clientset "github.com/nokia/danm/crd/client/clientset/versioned"
+    "github.com/nokia/danm/pkg/ipam"
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type calicoDeleteServiceImpl struct {}
+
+func (h *calicoDeleteServiceImpl) WasIPAllocatedByMe(danmClient clientset.Interface, ep *danmtypes.DanmEp, dnet *danmtypes.DanmNet) bool {
+    return ! ipam.WasIpAllocatedByDanm(ep.Spec.Iface.Address, dnet.Spec.Options.Cidr) &&
```

Comment:
hate to be that guy... but I think cleaning and checking should be per IP, rather than per DanmEp
I know unlikely scenario, but it is possible to use different IPAMs for different IPs in a dual-stack DanmEp right?
for those cases I think current logic would not work

Reply:
> I know unlikely scenario, but it is possible to use different IPAMs for different IPs in a dual-stack DanmEp right?

I did not know such case is even possible to configure with DANM. Please give an example configuration for such case and highlight the significant part where the magic happens.

Reply:
no, you are right! mixed IPAM in the same DanmEp is not possible :)
false alert